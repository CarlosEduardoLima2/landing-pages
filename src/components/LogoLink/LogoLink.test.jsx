import { screen } from '@testing-library/react';
import { LogoLink } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="logo" />);
    const heading = screen.getByRole('heading', { name: 'logo' });
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });

  it('should render image logo', () => {
    renderTheme(<LogoLink link="#target" text="logo" srcImg="img.png" />);
    const heading = screen.getByRole('heading', { name: 'logo' });
    expect(heading.firstChild.firstChild).toHaveAttribute('src', 'img.png');
  });

  it('should match snapshot', () => {
    renderTheme(<LogoLink link="#target" text="logo" srcImg="img.png" />);
    const heading = screen.getByRole('heading', { name: 'logo' });
    expect(heading.firstChild.firstChild).toMatchSnapshot();
  });
});
