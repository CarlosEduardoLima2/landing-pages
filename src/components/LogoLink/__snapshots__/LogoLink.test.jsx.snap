// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
<img
  alt="logo"
  src="img.png"
/>
`;
